### ISSUE TITLE
ESP32 don't need to output a MCLK signal to wm8978?

### ISSUE THREAD

**Issue by donny681** [GOLDELEC/Huan#7]
Hi,I watch your Schematics.The ESP32 does not output a MCLK signal.I refer the demos about stm32.The Stm32 chip output a 12M MCLK signal to wm8978.

**Comment by fanday** [comment]
@donny681 yes, this demo is not work.because there is no MCLK output from ESP32.orz!

**Comment by fanday** [comment]
@donny681 @GOLDELEC  ESP32 can output MCLK via GPIO0, below is the reference code:
        reg_val = REG_READ(PIN_CTRL);
	ESP_LOGI(TAG, "PIN_CTRL before:%x", reg_val);
	REG_WRITE(PIN_CTRL, 0xFFFFFFF0);
	reg_val = REG_READ(PIN_CTRL);
	ESP_LOGI(TAG, "PIN_CTRL after:%x", reg_val);
	PIN_FUNC_SELECT(GPIO_PIN_REG_0, 1); //GPIO0 as CLK_OUT1

**Comment by ghost** [comment]
Spotted this on io-bits.com -  might be helpful to anyone searching.
https://esp32.com/viewtopic.php?f=5&t=1585#p11715

If you want to output clock for I2S0 to
CLK_OUT1, then set PIN_CTRL[3:0] = 0x0
CLK_OUT2, then set PIN_CTRL[3:0] = 0x0 and PIN_CTRL[7:4] = 0x0
CLK_OUT3, then set PIN_CTRL[3:0] = 0x0 and PIN_CTRL[11:8] = 0x0

If you want to output clock for I2S1 to
CLK_OUT1, then set PIN_CTRL[3:0] = 0xF
CLK_OUT2, then set PIN_CTRL[3:0] = 0xF and PIN_CTRL[7:4] = 0x0
CLK_OUT3, then set PIN_CTRL[3:0] = 0xF and PIN_CTRL[11:8] = 0x0

The rest of the bits are not documented, so should be treated as reserved.

REG_WRITE(PIN_CTRL, 0b111111110000);
PIN_FUNC_SELECT(PERIPHS_IO_MUX_GPIO0_U, FUNC_GPIO0_CLK_OUT1);
